### PR TITLE
Add docs for new API endpoints

### PR DESCRIPTION
**Pull request by lucasmoura**
## Why is this needed?
Add documentation for the following API related changes:

* u.pro.detach.v1 endpoint
* u.pro.attach.token.full_token_attach.v1 endpoint
* u.pro.status.is_attached.v1 new fields in the API result

## Test Steps
Verify if the documentation page is correctly rendered and we are not missing any information related to the API changes

## Checklist

 - [ ] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] Changes here need to be documented, and this was done in: 

## Does this PR require extra reviews?

 - [ ] Yes
 - [ ] No

### CODE REVIEW

File: docs/references/api.rst [hunk 1]
```diff
@@ -631,6 +633,148 @@ like ``systemctl start ubuntu-advantage.service``.
...
+u.pro.attach.token.full_token_attach.v1
+============================================
+
+This endpoint allow the user to attach to a Pro subscription using
+a token.
+
+- Introduced in Ubuntu Pro Client Version: ``32~``
+- Args:
+
+  - ``token``: The token associated with a Pro subscriptionn
```
Comment (s-makin) [inline docs/references/api.rst:646]:
```suggestion
  - ``token``: The token associated with a Pro subscription
```

File: docs/references/api.rst [hunk 2]
```diff
@@ -631,6 +633,148 @@ like ``systemctl start ubuntu-advantage.service``.
...
+
+             * - Field Name
+               - Type
+               - Description
+             * - ``enabled``
+               - *List[str]*
+               - The services enabled during the attach operation
+             * - ``reboot_required``
+               - *bool*
+               - If the system requires a reboot after the attach operation
```
Comment (s-makin) [inline docs/references/api.rst:679]: Presumably this is True if the system requires a reboot?

File: docs/references/api.rst [hunk 3]
```diff
@@ -631,6 +633,148 @@ like ``systemctl start ubuntu-advantage.service``.
...
+   .. tab-item:: CLI interaction
+      :sync: CLI
+
+      - Calling from the CLI:
+
+        .. code-block:: bash
+
+           pro api u.pro.attach.token.full_token_attach.v1
+
+        Note that we don't need to explicit pass the token when calling the CLI command.
```
Comment (s-makin) [inline docs/references/api.rst:694]:
```suggestion
        Note that we don't need to explicitly pass the token when calling the CLI command.
```

File: docs/references/api.rst [hunk 4]
```diff
@@ -631,6 +633,148 @@ like ``systemctl start ubuntu-advantage.service``.
...
+        If we define a json file (i.e. file.json) with the same attributes as the options for this endpoint:
```
Comment (s-makin) [inline docs/references/api.rst:695]:
```suggestion
        If we define a JSON file (i.e. ``file.json``) with the same attributes as the options for this endpoint:
```

File: docs/references/api.rst [hunk 5]
```diff
@@ -631,6 +633,148 @@ like ``systemctl start ubuntu-advantage.service``.
...
+
+        .. code-block:: json
+
+           {
+               "token": "TOKEN",
+               "auto_enable_services": false
+           }
+
+        Then, we can call the API like this:
```
Comment (s-makin) [inline docs/references/api.rst:704]:
```suggestion
        Then we can call the API like this:
```

File: docs/references/api.rst [hunk 6]
```diff
@@ -631,6 +633,148 @@ like ``systemctl start ubuntu-advantage.service``.
...
+
+             * - Field Name
+               - Type
+               - Description
+             * - ``disabled``
+               - *List[str]*
+               - The services disabled during the detach operation
+             * - ``reboot_required``
+               - *bool*
+               - If the system requires a reboot after the detach operation
```
Comment (s-makin) [inline docs/references/api.rst:754]: Should this also be "True if..."

File: docs/references/api.rst [hunk 7]
```diff
@@ -2228,6 +2382,17 @@ This endpoint shows if the machine is attached to a Pro subscription.
         .. code-block:: bash
 
            pro api u.pro.status.is_attached.v1
+           
+   .. tab-item:: Explanation
+      :sync: explanation
+
+      The *contract_status* field can return 4 different states, they are:
```
Comment (s-makin) [inline docs/references/api.rst:2389]:
```suggestion
      The ``contract_status`` field can return 4 different states, they are:
```

File: docs/references/api.rst [hunk 8]
```diff
@@ -2228,6 +2382,17 @@ This endpoint shows if the machine is attached to a Pro subscription.
...
+
+      * **active**: The contract is currently valid.
+      * **grace-period**: The contract is in the grace period. This means that it is expired,
+        there are still some days were the contract will be valid.
```
Comment (s-makin) [inline docs/references/api.rst:2393]:
```suggestion
      * **grace-period**: The contract is in the grace period. This means that it is expired,
        but there are still some days where the contract will be valid.
```

**Review by s-makin — COMMENTED** [review]
Just a few nits :)

File: docs/references/api.rst (hunk 2 above)
Comment (lucasmoura) [reply docs/references/api.rst:679]: Yes, do you think we should make this explicit ?

Comment (lucasmoura) [reply docs/references/api.rst:679]: Yes, I will make it more explicit

**Comment by lucasmoura** [comment]
@s-makin updated

File: docs/references/api.rst [hunk 9]
```diff
@@ -631,6 +633,148 @@ like ``systemctl start ubuntu-advantage.service``.
...
+
+             * - Field Name
+               - Type
+               - Description
+             * - ``enabled``
+               - *List[str]*
+               - The services enabled during the attach operation
+             * - ``reboot_required``
+               - *bool*
+               - True If the system requires a reboot after the attach operation
```
Comment (s-makin) [inline docs/references/api.rst:679]:
```suggestion
               - True if the system requires a reboot after the attach operation
```

File: docs/references/api.rst [hunk 10]
```diff
@@ -2219,6 +2363,16 @@ This endpoint shows if the machine is attached to a Pro subscription.
              * - ``is_attached``
                - *bool*
                - If the machine is attached to a Pro subscription
+             * - ``contract_status``
+               - *str*
+               - The current contract status (active, grace-period, active-soon-to-expire, expired).
+                 Please refer to the explanation tab for a description of each state
```
Comment (s-makin) [inline docs/references/api.rst:2369]:
```suggestion
                 Please refer to the explanation tab for a description of each state.
```

**Review by s-makin — APPROVED** [review]
Just two little nits

**Comment by lucasmoura** [comment]
@s-makin fixed

**Review by s-makin — APPROVED** [review]

**Pull request merged**